### PR TITLE
QG-6 - WIP

### PR DESCRIPTION
**Pull request by nvminhtam**
Login with Google API.
Haven't implemented logic yet.

### CODE REVIEW

File: src/controllers/auth.controller.ts [hunk 1]
```diff
@@ -77,4 +77,59 @@ const login = async (req: Request, res: Response) => {
...
+    const user = await findUser({ username , email });
+    if (user) {
+      return res.json({
+        success: true,
+        message: 'Log in successfully again.',
+        accessToken,
+        expiresIn,
+      });
+    }
+    await createUser(decoded.name, email, "0912345678", username as string, username as string);
```
Comment (huy27201) [inline src/controllers/auth.controller.ts]: What is `0912345678` ?

Comment (nvminhtam) [reply src/controllers/auth.controller.ts]: Google OAuth does not return phone number so it's just a random number

Comment (huy27201) [reply src/controllers/auth.controller.ts]: Can we set it `null`?

Comment (nvminhtam) [reply src/controllers/auth.controller.ts]: Yes

**Review by huy27201 — APPROVED** [review]

**Pull request merged**